core/tests/mod.rs

```diff
@@ -0,0 +1,778 @@
+                    .expect("Failed to deserialize genesis block from file");
+                *cache = Some(gen.clone());
+                gen
+            }
+        };
+        
+        let rose_definition_id = <AssetDefinition as Identifiable>::Id::from_str("rose#wonderland")
+            .expect("valid names");
+        let alice_id =
+            <Account as Identifiable>::Id::from_str("alice@wonderland").expect("valid names");
+
+        let mint_rose_permission = PermissionToken::new(
+            "can_mint_assets_with_definition"
+                .parse()
+                .expect("valid names"),
+        )
+        .with_params([(
+            "asset_definition_id".parse().expect("valid names"),
+            IdBox::from(rose_definition_id.clone()).into(),
+        )]);
+        let burn_rose_permission = PermissionToken::new(
+            "can_burn_assets_with_definition"
+                .parse()
+                .expect("valid names"),
+        )
+        .with_params([(
+            "asset_definition_id".parse().expect("valid names"),
+            IdBox::from(rose_definition_id).into(),
+        )]);
+        let unregister_any_peer_permission =
+            PermissionToken::new("can_unregister_any_peer".parse().expect("valid names"));
+        let unregister_any_role_permission =
+            PermissionToken::new("can_unregister_any_role".parse().expect("valid names"));
+        let upgrade_validator_permission =
+            PermissionToken::new("can_upgrade_validator".parse().expect("valid names"));
+
+        for permission in [
+            mint_rose_permission,
+            burn_rose_permission,
+            unregister_any_peer_permission,
+            unregister_any_role_permission,
+            upgrade_validator_permission,
+        ] {
+            genesis.transactions[0]
+                .isi
+                .push(GrantBox::new(permission, alice_id.clone()).into());
+        }
+        
+        GenesisNetwork::from_configuration(
+            submit_genesis,
+            genesis,
+            Some(&cfg.genesis),
+            &cfg.wsv.transaction_limits,
+        )
+        .expect("Failed to init genesis")
+    }
+}
+
+#[test]
+fn sumeragi_commits_and_broadcasts_genesis() {
+    let keypair = KeyPair::generate().unwrap();
+    let (public_key, private_key) = keypair.clone().into();
+
+    let other_key = KeyPair::generate().unwrap();
+    let peer_ids = vec![PeerId::new(&"localhost:0".parse().unwrap(), &public_key), PeerId::new(&"localhost:1".parse().unwrap(), other_key.public_key())];
+
+    let mut configuration = {
+        let mut sample_proxy =
+            iroha::samples::get_config_proxy(peer_ids.iter().cloned().collect(), Some(keypair.clone()));
+        let env_proxy = ConfigurationProxy::from_env();
+        
+        sample_proxy.override_with(env_proxy)
+                    .build()
+                    .expect("Test Iroha config failed to build. This is either a programmer error or a compiler bug.")
```

Comment:
Not sure it worth mentioning that it could be a compiler bug. At the end we don't write it in other panic messages. Have we ever encountered such a bug?

Reply:
Well compiler bugs are not ucnommon, but they are hardly easily reproducible, so outright saying something is a compiler bug is just disrespectful of the `rustc` team.

Reply:
Can this conversation be closed?